shared/modules/caip-stream.ts

```diff
@@ -65,9 +65,10 @@ export class CaipStream extends Duplex {
 export const createCaipStream = (portStream: Duplex): Duplex => {
   const caipStream = new CaipStream();
 
-  pipeline(portStream, caipStream, portStream, (err: Error) =>
-    console.log('MetaMask CAIP stream', err),
-  );
+  pipeline(portStream, caipStream, portStream, (err: Error) => {
+    caipStream.substream.destroy();
```

Comment:
Is this related to other changes in this PR? Or is this a bug we found along the way?

Reply:
I suspect this is unrelated. I will try submitting this as a separate PR.

Reply:
Maybe this is okay to leave in here... it's just a small thing.

Reply:
Yes, I think the test above was added to cover this line.